src/topology/ping_pong.rs

```diff
@@ -0,0 +1,865 @@
+    use std::io::Cursor;
+
+    use super::*;
+    use crate::vdaf::dummy;
+    use assert_matches::assert_matches;
+
+    #[test]
+    fn ping_pong_one_round() {
+        let verify_key = [];
+        let aggregation_param = dummy::AggregationParam(0);
+        let nonce = [0; 16];
+        #[allow(clippy::let_unit_value)]
+        let public_share = ();
+        let input_share = dummy::InputShare(0);
+
+        let leader = dummy::Vdaf::new(1);
+        let helper = dummy::Vdaf::new(1);
+
+        // Leader inits into round 0
+        let (leader_state, leader_message) = leader
+            .leader_initialize(
+                &verify_key,
+                &aggregation_param,
+                &nonce,
+                &public_share,
+                &input_share,
+            )
+            .unwrap();
+
+        // Helper inits into round 1
+        let (helper_state, helper_message) = helper
+            .helper_initialize(
+                &verify_key,
+                &aggregation_param,
+                &nonce,
+                &public_share,
+                &input_share,
+                &leader_message,
+            )
+            .unwrap()
+            .evaluate(&helper)
+            .unwrap();
+
+        // 1 round VDAF: helper should finish immediately.
+        assert_matches!(helper_state, State::Finished(_));
+
+        let leader_state = leader
+            .continued(Role::Leader, leader_state, &helper_message)
+            .unwrap();
+        // 1 round VDAF: leader should finish when it gets helper message and emit no message.
+        assert_matches!(leader_state, ContinuedValue::FinishedNoMessage { .. });
+    }
+
+    #[test]
+    fn ping_pong_two_rounds() {
+        let verify_key = [];
+        let aggregation_param = dummy::AggregationParam(0);
+        let nonce = [0; 16];
+        #[allow(clippy::let_unit_value)]
+        let public_share = ();
+        let input_share = dummy::InputShare(0);
+
+        let leader = dummy::Vdaf::new(2);
+        let helper = dummy::Vdaf::new(2);
+
+        // Leader inits into round 0
+        let (leader_state, leader_message) = leader
+            .leader_initialize(
+                &verify_key,
+                &aggregation_param,
+                &nonce,
+                &public_share,
+                &input_share,
+            )
+            .unwrap();
+
+        // Helper inits into round 1
+        let (helper_state, helper_message) = helper
+            .helper_initialize(
+                &verify_key,
+                &aggregation_param,
+                &nonce,
+                &public_share,
+                &input_share,
+                &leader_message,
+            )
+            .unwrap()
+            .evaluate(&helper)
+            .unwrap();
+
+        // 2 round VDAF, round 1: helper should continue.
+        assert_matches!(helper_state, State::Continued(_));
+
+        let leader_state = leader
+            .continued(Role::Leader, leader_state, &helper_message)
```

Comment:
What about "initialize" and _"continue"_? Seems odd to have different parts of speech.

Reply:
We could also update the spec, if you'd like to resolve the inconsistency that way!

Reply:
Chris and I discussed offline and agreed that it's OK for the spec to say `leader_init` and this impl to say `leader_initialized`; conventions on abbreviations differ across languages and that's fine.